fe/fe-core/src/main/java/com/starrocks/catalog/HivePartitionKey.java

```diff
@@ -29,4 +31,34 @@ public HivePartitionKey() {
     public List<String> nullPartitionValueList() {
         return ImmutableList.of(HiveMetaClient.PARTITION_NULL_VALUE);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        PartitionKey that = (PartitionKey) o;
+
+        if (getKeys().size() != that.getKeys().size()) {
+            return false;
+        }
+
+        for (int i = 0; i < getKeys().size(); i++) {
+            LiteralExpr literalExpr = getKeys().get(i);
+            LiteralExpr thatLiteralExpr = that.getKeys().get(i);
+
+            // like hour=06 and hour=6, they are two partitions in hms
+            if (literalExpr instanceof IntLiteral && thatLiteralExpr instanceof IntLiteral) {
+                if (!literalExpr.getStringValue().equals(thatLiteralExpr.getStringValue())) {
+                    return false;
+                }
+            }
```

Comment:
Why do you not compare when they are not IntLiteral?
seems "2020-01-01" and "2022-01-01" will be considered equal

Reply:
because parent's `equals()` function will compare it.